### PR TITLE
Initial topbar with search functionality

### LINKED ISSUE #5: Fix top bar CSS and add more features (communicate with design)
Get the search debugged and add more icons as necessary, contact design for help on icons.

### PR DESCRIPTION
**Pull request by atar13**

### CODE REVIEW

File: client/src/homeComponents/topBarComponents/TopBar.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ * @param {*} courses 
+ * @param {*} query 
+ */
+const querySearch = (courses, query) => {
+    if (!query) {
+        return []
+    }
+    return courses.filter((course) => {
+        const courseName = course.code;
+        return courseName.includes(query.toUpperCase())
```
Comment (tristinxie) [inline client/src/homeComponents/topBarComponents/TopBar.js]: Have consistent semicolons throughout your code. Every line should have a semicolon.

File: client/src/homeComponents/topBarComponents/TopBar.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    useEffect(() => {
+        
+        if(showResults){
+            setHideResults("")
+        }
+
+    }, [showResults])
+
+
+    return (<div id="search-results-container" ref={resultsWrapperRef} >
```
Comment (tristinxie) [inline client/src/homeComponents/topBarComponents/TopBar.js]: keep return lines consistent in this style:
```
return (

)
```

File: client/src/homeComponents/topBarComponents/TopBar.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+
+
+    return (
+        //TODO: search bar 20% from the left and profile button on the far right
+        <div>
+            <div id="topbar-container">
+                <SearchBar updateQuery={updateQuery} />
+                <div className="topbar-buttons" style={{ marginRight: '2vw', paddingRight: '1vh', paddingLeft: '1vh' }}>
```
Comment (tristinxie) [inline client/src/homeComponents/topBarComponents/TopBar.js]: Move inline style to external css file

File: client/src/homeComponents/topBarComponents/TopBar.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    return courses.filter((course) => {
+        const courseName = course.code;
+        return courseName.includes(query.toUpperCase())
+    })
+}
+
+
+const SearchResults = (props) => {
+
+    const { query} = props
```
Comment (tristinxie) [inline client/src/homeComponents/topBarComponents/TopBar.js]: consistent spacing

**Review by tristinxie — CHANGES_REQUESTED** [review]
There are a few styling guidelines I want you to follow. Also, you should separate a lot of these components into separate files.

**Comment by atar13** [comment]
I moved each component to its own file. Should I also move each component's style to a seperate css file?

**Review by tristinxie — COMMENTED** [review]
Looks good so far! Request for approval again once you are satisfied with the functionality.

**Review by tristinxie — APPROVED** [review]

**Pull request merged**